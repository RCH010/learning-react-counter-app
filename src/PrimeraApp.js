import React from 'react';
import PropTypes from 'prop-types'
// import React, { Fragment } from 'react';
// functional componets

// const PrimeraApp = () => {
//     // con solo las etiquetas asi, las hace como si fuera un fragment
//     return (
//     <>
//         <h1>Hola Mundo</h1>
//         <p>Mi primera APP</p>
//     </>
//     );
// }
const PrimeraApp = ({saludo, subtitulo}) => {
    return (
    <>
        <h1>{saludo}</h1>
        {/* <pre>{JSON.stringify(saludo2,null,3)}</pre> */}
        <p>{subtitulo}</p>
    </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Bienvenido'
}

export default PrimeraApp;