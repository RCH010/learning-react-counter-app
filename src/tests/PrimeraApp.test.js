import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp/>', () => {
    test('should show message "msg"', () => {
        const saludo =  'Hola Mundo';
        const wrapper = render(<PrimeraApp saludo={saludo}/>);
        expect(wrapper.getByText(saludo)).toBeInTheDocument();
    })

    //  AHORA CON ENZYME
    test('should show <PrimeraApp /> correctly', () => {
        const saludo =  'Hola Mundo';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show subtitutle send thorugh props', (done) => {
        const saludo =  'Hola Mundo';
        const subtitulo = 'Prueba de Subtitulo';
        const wrapper = shallow(
        <PrimeraApp 
            saludo = {saludo}
            subtitulo = {subtitulo}
        />);
        expect(wrapper).toMatchSnapshot();
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo)
        done();
    })
    
})
// enzyme, es m√°s facil para hacer estas pruebas