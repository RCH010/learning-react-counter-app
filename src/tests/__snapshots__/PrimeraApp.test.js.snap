// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> should show <PrimeraApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <p>
    Bienvenido
  </p>
</Fragment>
`;

exports[`Pruebas en <PrimeraApp/> should show subtitutle send thorugh props 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <p>
    Prueba de Subtitulo
  </p>
</Fragment>
`;
